src/main/kotlin/io/ionic/libs/ionfileviewerlib/IONFLVWController.kt

```diff
@@ -0,0 +1,96 @@
+package io.ionic.libs.ionfileviewerlib
+
+import android.app.Activity
+import androidx.annotation.WorkerThread
+import io.ionic.libs.ionfileviewerlib.helpers.IONFLVWAssetsHelper
+import io.ionic.libs.ionfileviewerlib.helpers.IONFLVWInputsValidator
+import io.ionic.libs.ionfileviewerlib.helpers.IONFLVWOpenDocumentHelper
+import io.ionic.libs.ionfileviewerlib.helpers.runCatchingIONFLVWExceptions
+import io.ionic.libs.ionfileviewerlib.model.IONFLVWException
+import kotlin.concurrent.thread
+
+/**
+ * Entry point in IONFileViewerLib-Android
+ *
+ * Contains relevant methods for viewing files in Android.
+ */
+class IONFLVWController internal constructor(
+    private val inputsValidator: IONFLVWInputsValidator,
+    private val assetsHelper: IONFLVWAssetsHelper,
+    private val openDocumentHelper: IONFLVWOpenDocumentHelper
+) {
+    constructor() : this(
+        inputsValidator = IONFLVWInputsValidator(),
+        assetsHelper = IONFLVWAssetsHelper(),
+        openDocumentHelper = IONFLVWOpenDocumentHelper()
+    )
+
+    /**
+     * Opens a document from a local file path.
+     *
+     * @param activity the Android activity
+     * @param filePath the full path to the file
+     * @return A Kotlin [Result] Object:
+     *  - successful if the document was opened;
+     *  - failure if an error occurred, exception of type [IONFLVWException]
+     */
+    fun openDocumentFromLocalPath(
+        activity: Activity,
+        filePath: String
+    ): Result<Unit> = runCatchingIONFLVWExceptions {
+        if (!inputsValidator.isPathValid(filePath)) {
+            throw IONFLVWException.InvalidPath(filePath)
+        }
+        val filePathNoSpaces: String =
```

Comment:
Could this be replaced with the following, so that `toRegex()` is only called once?

`val filePathNoSpaces = filePath.replace("%20| ".toRegex(), "\\ ")`